tests/unit/cocoon/test_utilities.py

```diff
@@ -631,18 +646,23 @@ def test_file_type_checks(self):
 
     @parameterized.expand(
         [
-            ["S&PCreditRating(UK)", "SandPCreditRatingUK"],
-            ["Return%", "ReturnPercentage"],
-            ["Buy/Sell Indicator", "BuySellIndicator"],
-            ["balances.available", "balances_available"],
+            ["Explicitly known invalid character '&'", "S&PCreditRating(UK)", "SandPCreditRatingUK"],
+            ["Explicitly known invalid character '%'", "Return%", "ReturnPercentage"],
+            ["Explicitly known invalid character '.'", "balances.available", "balances_available"],
+            ["Invalid character not meeting regex #1", "Buy/Sell Indicator", "BuySellIndicator"],
+            ["Invalid character not meeting regex #2", "£DollarDollarBills$", "DollarDollarBills"],
+            ["Integer", 1, "1"],
+            ["Decimal", 1.8596, "1_8596"],
+            ["List", ["My", "List", "Code"], "MyListCode"]
         ]
```

Comment:
Added it in. The point is that we have moved to the below which means that we now have a whitelist instead of a blacklist so nothing can slip through the cracks!

    friendly_code = (
        re.sub(r'[^\w]', '', raw_code
            .replace("%", "Percentage")
            .replace("&", "and")
            .replace(".", "_")
            .replace("-", "_")
            .strip()
        )
    )